src/app/public/modules/text-editor/text-editor.component.ts

```diff
@@ -0,0 +1,333 @@
+import {
+  Component,
+  ViewEncapsulation,
+  forwardRef,
+  Input,
+  AfterViewInit,
+  ViewChild,
+  ElementRef,
+  OnDestroy,
+  ChangeDetectorRef,
+  ChangeDetectionStrategy
+} from '@angular/core';
+
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR
+} from '@angular/forms';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  MENUBAR_ACTION_DEFAULTS
+} from './defaults/menubar-section-defaults';
+
+import {
+  STYLE_STATE_DEFAULTS
+} from './defaults/style-state-defaults';
+
+import {
+  TOOLBAR_SECTION_DEFAULTS
+} from './defaults/toolbar-section-defaults';
+
+import {
+  SkyTextEditorService
+} from './services/text-editor.service';
+
+import {
+  SkyTextMergeFieldService
+} from './services/text-merge-field.service';
+
+import {
+  SkyTextSanitizationService
+} from './services/text-sanitization.service';
+
+import {
+  SkyTextSelectionManagementService
+} from './services/text-selection-management.service';
+
+import {
+  FONT_LIST_DEFAULTS
+} from './defaults/font-list-defaults';
+
+import {
+  FONT_SIZE_LIST_DEFAULTS
+} from './defaults/font-size-list-defaults';
+
+import {
+  SkyTextEditorFont
+} from './types/font-state';
+
+import {
+  SkyTextEditorMenubarAction
+} from './types/menubar-action';
+
+import {
+  SkyTextEditorStyleState
+} from './types/style-state';
+
+import {
+  SkyTextEditorMergeField
+} from './types/text-editor-merge-field';
+
+import {
+  SkyTextEditorToolbarActions
+} from './types/toolbar-action';
+
+/**
+ * Auto-incrementing integer used to generate unique ids for radio components.
+ */
+let nextUniqueId = 0;
+
+/**
+ * The text editor component lets users format and manipulate text.
+ */
+@Component({
+  selector: 'sky-text-editor',
+  templateUrl: './text-editor.component.html',
+  styleUrls: ['./text-editor.component.scss'],
+  providers: [
+    SkyTextEditorService,
+    SkyTextMergeFieldService,
+    SkyTextSelectionManagementService,
+    {
+      provide: NG_VALUE_ACCESSOR,
+      // tslint:disable-next-line: no-forward-ref
+      useExisting: forwardRef(() => SkyTextEditorComponent),
+      multi: true
+    }
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None
+})
+export class SkyTextEditorComponent implements AfterViewInit, ControlValueAccessor, OnDestroy {
```

Comment:
Do we need a `setDisabledState` function on this? I don't see any current way to disable the editor but it is a form accessor and so can be disabled via the form control.

Reply:
We would also potentially need a `disabled` input to allow template forms to disable.

Reply:
Good catch. Let's defer for beta, if you don't mind. Will create a bug for this.

Reply:
I'm fine with that. With this one being consumer facing it might be good to let our consumers know that this is known missing functionality. 

(Going to leave this one unresolved to make sure you see this - resolve once you do 😄)